plugins/amazonq/chat/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/amazonq/webview/BrowserConnectorTest.kt

```diff
@@ -0,0 +1,213 @@
+                )
+
+                assert(additionalMessages.isEmpty())
+                assert(issuesCaptor.firstValue.isNotEmpty())
+                assert(issuesCaptor.firstValue[0].title == "Test Issue")
+            }
+        }
+    }
+
+    @Test
+    fun `parseFindingsMessages should skip directory files and not populate mappedFindings`() {
+        val mockDirectoryFile = mock<VirtualFile> { on { isDirectory } doReturn true }
+
+        mockStatic(LocalFileSystem::class.java).use { localFileSystemMock ->
+            localFileSystemMock.`when`<LocalFileSystem> { LocalFileSystem.getInstance() } doReturn mockLocalFileSystem
+            whenever(mockLocalFileSystem.findFileByIoFile(any())) doReturn mockDirectoryFile
+
+            val issue = BrowserConnector.FlareCodeScanIssue(
+                startLine = 1, endLine = 1, comment = null, title = "Test Issue",
+                description = Description("Test description", "Test text"), detectorId = "test-detector",
+                detectorName = "Test Detector", findingId = "test-finding-id", ruleId = null,
+                relatedVulnerabilities = emptyList(), severity = "MEDIUM",
+                recommendation = Recommendation("Fix this", "https://example.com"), suggestedFixes = emptyList(),
+                scanJobId = "test-job-id", language = "kotlin", autoDetected = true,
+                filePath = "/test/directory", findingContext = "test context"
+            )
+
+            val aggregatedIssue = BrowserConnector.AggregatedCodeScanIssue("/test/directory", listOf(issue))
+            val findingsMessage = mapOf(
+                "messageId" to "test_displayFindings",
+                "body" to jacksonObjectMapper().writeValueAsString(listOf(aggregatedIssue))
+            )
+            val additionalMessages = mutableListOf<Map<String, Any>>(findingsMessage)
+            val messagesMap = mapOf("additionalMessages" to additionalMessages)
+
+            browserConnector.parseFindingsMessages(messagesMap)
+
+            val issuesCaptor = argumentCaptor<List<CodeWhispererCodeScanIssue>>()
+            verify(mockCodeScanManager).addOnDemandIssues(
+                issuesCaptor.capture(),
+                any(),
+                eq(CodeWhispererConstants.CodeAnalysisScope.AGENTIC)
+            )
+
+            assert(issuesCaptor.firstValue.isEmpty())
+            assert(additionalMessages.isEmpty())
+        }
+    }
+
+    @Test
+    fun `parseFindingsMessages should handle invalid JSON gracefully`() {
+        val findingsMessage = mapOf(
+            "messageId" to "test_codeReviewFindings",
+            "body" to "invalid json"
+        )
+        val additionalMessages = mutableListOf<Map<String, Any>>(findingsMessage)
+        val messagesMap = mapOf("additionalMessages" to additionalMessages)
+
+        browserConnector.parseFindingsMessages(messagesMap)
+
+        assert(additionalMessages.isEmpty())
```

Comment:
should be using fluent asserts via assertj
```suggestion
        assertThat(additionalMessages).isEmpty()
```